router/src/lib.rs

```diff
@@ -1221,9 +1223,15 @@ pub struct TextMessage {
 
 impl From<Message> for TextMessage {
     fn from(value: Message) -> Self {
+        let content = value
```

Comment:
Maybe we could add an additional check here to prevent this type of messages from anything other than the `assistant`, as well as if both `content` and `tool_calls` are provided, WDYT? Maybe even a custom new error as you suggested recently.

Reply:
good point! I've updated the PR to prefer a none optional enum that ensure either a content or tools are provided - looking into how errors may be improved now